### PR TITLE
[Flink 4599] - Add 'explain()' also to StreamTableEnvironment

### PR DESCRIPTION
**Pull request by chobeat**

### CODE REVIEW

**Comment by twalthr** [comment]
Could you rebase your PR on the current master and then do a force push? There are a lot of commits that should not appear here.

**Comment by chobeat** [comment]
Done, sorry.

File: docs/dev/table_api.md [hunk 1]
```diff
@@ -2457,3 +2457,27 @@ The Table API provides a configuration (the so-called `TableConfig`) to modify r
...
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+	val env = StreamExecutionEnvironment.getExecutionEnvironment
+	val tEnv = TableEnvironment.getTableEnvironment(env)
+
+	val table1 = env.fromElements((1, "hello")).toTable(tEnv, 'count, 'word)
+	val table2 = env.fromElements((1, "hello")).toTable(tEnv, 'count, 'word)
+	val table = table1.unionAll(table2)
+
+	val explanation:String = tEnv.explain(table)
```
Comment (apivovarov) [inline docs/dev/table_api.md]: put space before String

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/StreamTableEnvironment.scala [hunk 2]
```diff
@@ -271,5 +271,21 @@ abstract class StreamTableEnvironment(
...
+  * @param table The table for which the AST and execution plan will be returned.
+    * @param extended Flag to include detailed optimizer estimates.
+  */
+   def explain(table: Table): String = {
+
+    val ast = RelOptUtil.toString(table.getRelNode)
+
+    s"== Abstract Syntax Tree ==" +
+      System.lineSeparator +
+      s"$ast"
```
Comment (apivovarov) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/StreamTableEnvironment.scala]: Maybe this? 
```
s"== Abstract Syntax Tree ==${System.lineSeparator}$ast"
```

Comment (apivovarov) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/StreamTableEnvironment.scala]: No need for s on line 285

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/stream/ExplainStreamTest.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      .filter("a % 2 = 0")
+
+    val result = tEnv.explain(table).replaceAll("\\r\\n", "\n")
+    val source = scala.io.Source.fromFile(testFilePath +
+      "../../src/test/scala/resources/testFilterStream0.out").mkString.replaceAll("\\r\\n", "\n")
+    assertEquals(result, source)
+  }
+
+  @Test
+  def testUnion() : Unit = {
```
Comment (apivovarov) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/stream/ExplainStreamTest.scala]: no need for space after () on line 35 and 50

**Comment by twalthr** [comment]
Thanks @chobeat. I fixed minor formatting issues and extended the documentation a little bit. Will merge...

**Pull request closed without merging**